zebra-network/src/peer_set/initialize/tests/vectors.rs

```diff
@@ -1161,6 +1162,51 @@ async fn add_initial_peers_is_rate_limited() {
     );
 }
 
+/// Test that [`init`] does not deadlock.
+#[tokio::test]
+async fn network_init_deadlock() {
+    // The `PEER_COUNT` is the amount of initial seed peers. The value is set so
+    // that the peers fill up `PEERSET_INITIAL_TARGET_SIZE`, fill up the channel
+    // for sending unused peers to the `AddressBook`, and so that there are
+    // still some extra peers left.
+    const PEER_COUNT: usize = 200;
+    const PEERSET_INITIAL_TARGET_SIZE: usize = 2;
+
+    // This should be a const, but [`Duration::new`] is not yet stable as a
+    // const fn.
+    let time_limit: Duration = Duration::new(10, 0);
```

Comment:
I think you can use `Duration::from_secs(10)`, which is a `const fn`.

```suggestion
    const TIME_LIMIT: Duration = Duration::from_secs(10);
```